pkg/java/pom/parse.go

```diff
@@ -409,7 +433,7 @@ func (p *parser) mergeDependencies(parent, child []artifact, exclusions map[stri
 	var deps []artifact
 	unique := map[string]struct{}{}
 
-	for _, d := range append(parent, child...) {
+	for _, d := range append(child, parent...) {
```

Comment:
https://github.com/aquasecurity/go-dep-parser/blob/c95688d311ba60dfada97294d7461f07b9602521/pkg/java/pom/parse.go#L325-L328

Looks like it was a mistake.
I've updated test to show this:
https://github.com/aquasecurity/go-dep-parser/pull/287/files#diff-54393967b4dbd54d8dab7c5bf81ff4297c6fb825ff2bbf1805a4fdabec231146R44